src/common/components/UserListItemBox.tsx

```diff
@@ -0,0 +1,16 @@
+import { Card, type CardProps } from "@radix-ui/themes";
+import type { ReactNode } from "react";
+
+interface UserBoxProps {
+  children: ReactNode;
+}
+
+type Props = UserBoxProps & CardProps;
+
+export default function UserListItemBox({ children, ...props }: Props) {
```

Comment:
* object type을 교차 시켜 사용할 때는 type intersection보다 interface extends를 사용하는 것이 성능 측면에서 이점이 있어요! [[ > 참고 링크]](https://github.com/microsoft/TypeScript/wiki/Performance#preferring-interfaces-over-intersections)
  * 이유  : type은 컴파일러에 의한 캐싱이 불가능하지만, interface는 가능 / interface extends 사용 시 declaration merge가 가능
* 이 부분은 컨벤션으로 정해놓지는 않았지만, 공통 컴포넌트는 타입을 확장시켜 사용할 가능성(ex. styled component 등으로 한 번 더 wrapping 해서 사용 가능)이 있기 때문에, interface extends로 prop type을 확장시키는 것이 성능 측면에서 좋다고 생각해요 ~! 참고하시와용

```suggestion
interface UserBoxProps extends CardProps {
  children: ReactNode;
}

export default function UserListItemBox({ children, ...props }: UserBoxProps) 
```

Reply:
> common/types/util.ts 에 두면 어떨까요? 🤔

저는 좋습니다~!

Reply:
그런데 생각해보니 `StrictPropsWithChildren`을 쓰게 되면 결국 children과는 intersection을 써서 type으로 쓰게 되는건데 위에 extends와 intersection에 대해 논의 나눈 것과 반대되는 결론인 것 같아요. 이 부분은 어떻게 생각하세요?

그런 의미에서 저는 다시 생각해봤을 때 children을 직접 선언해주는 것이 interface를 쓰는 점에서도, 코드를 이해하는 측면에서도 오히려 직관적이라 더 나은 방식일 수 있겠다는 생각이 드네요!

Reply:
어찌보면 StrictPropsWithChildren 방식도 직관성 측면에선 나쁘지 않다는 생각이 드는데, 사실 제 개인적인 취향은 children을 직접 선언해주는게 어떤 prop이 필요한지 직관적으로 드러나서 좋긴 하더라구요!!!

근데 이 부분은 뭐가 명확히 좋다라기보단 컨벤션 나름이라는 생각이 들어서 연서 생각도 들어보고 정하면 좋을 것 같네용 😃

Reply:
저도 children을 직접 선언 해주는 것이 직관적으로 드러나서 좋다는 말에 동의해요 ~!
저는 children prop만 사용될 때는 편의상 PropsWithChildren으로 선언해주는 것을 선호하는데, 현재는 해당 패턴이 자주 반복되지 않아 공통 유틸 타입으로 뺐을 때의 이점이 크지는 않은것 같아서, children을 직접 선언해서 사용하는 방식도 좋아요!!

Reply:
연서 말대로 children prop만 사용되는 패턴이 많이 반복될 경우에는 공통 유틸 타입으로 빼서 사용하는 게 더 편리할 것 같네요! 연서 덕분에 좋은 거 또 하나 배워갑니당 🚀🚀